packages/bundler-plugin-core/src/options-mapping.ts

```diff
@@ -52,57 +57,84 @@ export type InternalIncludeEntry = RequiredInternalIncludeEntry &
   };
 
 export function normalizeUserOptions(userOptions: UserOptions): InternalOptions {
-  let entries: (string | RegExp)[] | ((filePath: string) => boolean) | undefined;
-  if (userOptions.entries === undefined) {
-    entries = undefined;
-  } else if (typeof userOptions.entries === "function" || Array.isArray(userOptions.entries)) {
-    entries = userOptions.entries;
+  const entries = normalizeEntries(userOptions.entries);
+  const include = normalizeInclude(userOptions);
+
+  return {
+    // Strictly required options
+    include,
+
+    // These options must be set b/c we need them for release injection
+    // They can also be set as environment variables. Technically, they
+    // could be set in the config file but this would be too late for
+    // release injection
+    org: userOptions.org ?? process.env["SENTRY_ORG"],
+    project: userOptions.project ?? process.env["SENTRY_PROJECT"],
+    release: userOptions.release ?? process.env["SENTRY_RELEASE"] ?? "",
+
+    // These options and can also be set via env variables or config file
+    // but are only passed to Sentry CLI. They could remain undefined
+    // because the config file is only read by the CLI
+    authToken: userOptions.authToken ?? process.env["SENTRY_AUTH_TOKEN"],
+    customHeader: userOptions.customHeader ?? process.env["CUSTOM_HEADER"],
+    url: userOptions.url ?? process.env["SENTRY_URL"] ?? "https://sentry.io/",
+    vcsRemote: userOptions.vcsRemote ?? process.env["SENTRY_VCS_REMOTE"] ?? "origin",
+
+    // Options with default values
+    finalize: userOptions.finalize ?? true,
+    cleanArtifacts: userOptions.cleanArtifacts ?? false,
+    dryRun: userOptions.dryRun ?? false,
+    debug: userOptions.debug ?? false,
+    silent: userOptions.silent ?? false,
+    telemetry: userOptions.telemetry ?? true,
+    injectReleasesMap: userOptions.injectReleasesMap ?? false,
+
+    // Optional options
+    setCommits: userOptions.setCommits,
+    deploy: userOptions.deploy,
+    entries,
+    dist: userOptions.dist,
+    errorHandler: userOptions.errorHandler,
+    configFile: userOptions.configFile,
+  };
+}
+
+/**
+ * Converts the user-facing `entries` option to the internal `entries` option
+ */
+function normalizeEntries(
+  userEntries: UserOptions["entries"]
+): (string | RegExp)[] | ((filePath: string) => boolean) | undefined {
+  if (userEntries === undefined) {
+    return undefined;
+  } else if (typeof userEntries === "function" || Array.isArray(userEntries)) {
+    return userEntries;
   } else {
-    entries = [userOptions.entries];
+    return [userEntries];
   }
+}
+
+function normalizeInclude(userOptions: UserOptions): InternalIncludeEntry[] {
```

Comment:
L: It might be nice to mention in a docstring here the different formats `include` can take.

Reply:
[85ce2a8](https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/103/commits/85ce2a8b507e4245dcd995cf816b8c9275f26c23)